File: app/src/main/java/com/sumian/sleepdoctor/pager/activity/WelcomeActivity.java
package com.sumian.sleepdoctor.pager.activity;

import android.view.View;

import com.sumian.sleepdoctor.R;
import com.sumian.sleepdoctor.account.userProfile.ImproveUserProfileOneActivity;
import com.sumian.sleepdoctor.account.login.LoginActivity;
import com.sumian.sleepdoctor.app.AppManager;
import com.sumian.sleepdoctor.app.delegate.OtherDelegate;
import com.sumian.sleepdoctor.base.BaseActivity;
import com.sumian.sleepdoctor.main.MainActivity;

/**
 * Created by jzz
 * on 2018/1/16.
 * desc:
 */

public class WelcomeActivity extends BaseActivity implements OtherDelegate {

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_pager_welcome;
    }

    @Override
    protected void initWidget(View root) {
        super.initWidget(root);
        setStatusBar();
    }

    @Override
    protected void initData() {
        super.initData();
        AppManager.getAccountViewModel().getLiveDataToken().observe(this, token -> mRoot.postDelayed(() -> {
            if (token == null) {
                LoginActivity.show(WelcomeActivity.this, LoginActivity.class);
            } else if (token.is_new) {
                ImproveUserProfileOneActivity.show(WelcomeActivity.this, ImproveUserProfileOneActivity.class);
            } else {
                MainActivity.show(WelcomeActivity.this, MainActivity.class);
            }
            finish();
        }, 50));
    }
}

File: patient/src/main/java/com/sumian/sd/main/WelcomeActivity.kt
package com.sumian.sd.main

import android.graphics.Color
import android.os.Handler
import android.util.Log
import com.sumian.common.base.BaseActivity
import com.sumian.sd.BuildConfig
import com.sumian.sd.R
import com.sumian.sd.app.AppManager
import com.sumian.sd.buz.account.login.LoginActivity
import com.sumian.sd.common.utils.StatusBarUtil
import com.sumian.sd.examine.login.ExamineLoginRouterActivity

/**
 * Created by jzz
 * on 2017/9/30
 *
 *
 * desc:
 */

class WelcomeActivity : BaseActivity() {

    companion object {
        private const val SPLASH_DURATION = 500L
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main_welcome
    }

    override fun initWidget() {
        super.initWidget()
        StatusBarUtil.setStatusBarTextColorDark(this, true)
        if (BuildConfig.IS_EXAMINE_VERSION) {
            window.decorView.setBackgroundColor(Color.BLACK)
        }
    }

    override fun initData() {
        super.initData()
        Handler().postDelayed({
            val login = AppManager.getAccountViewModel().isLogin
            if (login) {
                AppManager.launchMain()
            } else {
                if (BuildConfig.IS_EXAMINE_VERSION) {
                    ExamineLoginRouterActivity.show(this)
                } else {
                    LoginActivity.show()
                }
            }
        }, SPLASH_DURATION)
    }

    override fun onStop() {
        super.onStop()
        // 之所以在onStop finish，是因为在启动MainActivity 时马上finish 会导致无法执行Activity转场动画，会显示APP下面的内容。
        finish()
    }

    override fun onBackPressed() {
        //super.onBackPressed();
    }


}
